monai/networks/blocks/convolutions.py

```diff
@@ -316,3 +317,37 @@ def forward(self, x: torch.Tensor) -> torch.Tensor:
         res: torch.Tensor = self.residual(x)  # create the additive residual from x
         cx: torch.Tensor = self.conv(x)  # apply x to sequence of operations
         return cx + res  # add the residual to the output
+
+
+class ClassifierBlock(Convolution):
+    """
+    Returns a classifier block without an activation function at the top.
+    It consists of a 1 * 1 convolutional layer which maps the input to a num_class channel feature map.
+    The output is a probability map for each of the classes.
+
+    Args:
+        spatial_dims: number of spatial dimensions.
+        in_channels: number of input channels.
+        out_channels: number of classes to map to.
+        strides: convolution stride. Defaults to 1.
+        kernel_size: convolution kernel size. Defaults to 3.
+        adn_ordering: a string representing the ordering of activation, normalization, and dropout.
+        Defaults to "NDA".
+        act: activation type and arguments. Defaults to PReLU.
+
+    """
+
+    def __init__(self, spatial_dims, in_channels, out_channels, strides, kernel_size, act=None, adn_ordering="A"):
+        super().__init__(spatial_dims, in_channels, out_channels, strides, kernel_size, adn_ordering, act)
+
+    def forward(self, input: torch.Tensor, weights=None, indices=None):
+        _, channel, _, _ = input.size()
+        if weights is not None:
+            weights, _ = torch.max(weights, dim=0)
+            weights = weights.view(1, channel, 1, 1)
+            # use weights to adapt how the classes are weighted.
+            out_conv = F.conv2d(input, weights)
```

Comment:
This is going to restrict this class to be 2D only, instead you can look at how many dims you have and choose function from that:

```suggestion
        _, channel, *dims = input.size()
        if weights is not None:
            weights, _ = torch.max(weights, dim=0)
            weights = weights.view(1, channel, 1, 1)
            # use weights to adapt how the classes are weighted.
            if len(dims)==2:
                out_conv = F.conv2d(input, weights)
            else:
                out_conv = F.conv3d(input, weights)
```

This does need to be checked to work and be compatible with Torchscript.

Reply:
If this really should only be for 2D that's ok and never mind the change but please do mention that in the docstring.

Reply:
We incorporate part of your suggestion, and we verify the user ask for a 2D architecture